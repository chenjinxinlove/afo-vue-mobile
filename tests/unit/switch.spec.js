import { mount } from '@vue/test-utils'
import Demo from '../../src/switch/demo'
import Vue from 'vue'
import Switch from '../../src/switch'

describe('Switch.vue', () => {
  // 测试是否可以被使用
  it('use', () => {
    Vue.use(Switch)
    expect(Vue.component(Switch.name)).toHaveProperty('name', 'VueComponent')
  })
  // demo测试快照
  it('demo snapshot', () => {
    expect(mount(Demo)).toMatchSnapshot()
  })
})