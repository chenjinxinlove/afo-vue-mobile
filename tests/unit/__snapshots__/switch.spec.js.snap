// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch.vue demo snapshot 1`] = `
<div>
  <p>基础用法</p>
  <div class="afo-switch">
    <input type="checkbox" class="afo-switch__input">
    <i class="afo-switch__ui"></i> <span class="afo-switch__label"></span></div>
  <div style="height: 20px;"></div>
  <p>禁用状态</p>
  <div class="afo-switch">
    <input type="checkbox" disabled="disabled" class="afo-switch__input">
    <i class="afo-switch__ui"></i> <span class="afo-switch__label"></span></div>
</div>
`;
